sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerConfig.java

```diff
@@ -277,6 +277,29 @@
    */
   private static final AtomicBoolean PRINTED_BUILD_INFO = new AtomicBoolean(false);
 
+  /**
+   * When defined, this is the role to use when subscribing to Mesos and for recovering or adding footprint.
+   */
+  private static final String MESOS_ALLOCATION_ROLE_ENV = "MESOS_ALLOCATION_ROLE";
+
+  /**
+   * When set to true, the {@link #MESOS_ALLOCATION_ROLE} value should be used to subscribe and recover or
+   * add footprint.
+   */
+  private static final String MARATHON_APP_ENFORCE_GROUP_ROLE_ENV = "MARATHON_APP_ENFORCE_GROUP_ROLE";
+
+  /**
+   * Default role that the scheduler gets created under via Marathon without any other options.
+   */
+  private static final String DEFAULT_SCHEDULER_ROLE = "slave_public";
+
+  /*
+   * Determines if the scheduler should subscribe with both legacy role and new quota roles.
+   * Setting this to true is harmless and might incur a slight startup penalty as Mesos offers
+   * with this role get declined if the pod is looking for role set by FrameworkConfig.getRole()
```

Comment:
change to javadoc and use `{@link com.mesosphere.sdk.framework.FrameworkConfig#getRole()}`